### PR TITLE
handling of join queries without WHERE/ON clauses

### PR DESCRIPTION
**Pull request by jgaleotti**

### CODE REVIEW

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package org.evomaster.client.java.sql;
+
+import java.util.TreeMap;
+
+public class QueryResultSet {
```
Comment (arcuri82) [inline client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java:5]: add JavaDoc, specifying what this class is for

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    private final TreeMap<String, QueryResult> queryResults;
```
Comment (arcuri82) [inline client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java:7]: any specific reason to use a TreeMap? even if the concrete class is going to be a TreeMap, shouldn't the reference be a Map? also, need to add comment specifying what the String keys are

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java [hunk 3]
```diff
@@ -3,28 +3,96 @@
...
-import static org.evomaster.client.java.sql.internal.SqlParserUtils.getWhere;
+import java.util.Collection;
+import java.util.List;
+
+import static org.evomaster.client.java.sql.internal.SqlParserUtils.*;
 
 public class SqlHeuristicsCalculator {
 
+    public static double C = 0.1;
+    public static double C_BETTER = C + C / 2;
```
Comment (arcuri82) [inline client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java:21]: put parentheses... i never remember which has precednce... is this equivalent to `C + (C/2)` ?

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java [hunk 4]
```diff
@@ -3,28 +3,96 @@
...
+    public static Truthness TRUE_TRUTHNESS = new Truthness(1, C);
+
+    private QueryResultSet queryResultSet;
+
+    private SqlHeuristicsCalculator(QueryResult[] data) {
+        final boolean isCaseSensitive = false;
```
Comment (arcuri82) [inline client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java:27]: a final boolean given as input? why not passing `false` directly? is this for readability?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java (hunk 3 above)
Comment (jgaleotti) [reply client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java:21]: ok

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java (hunk 4 above)
Comment (jgaleotti) [reply client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlHeuristicsCalculator.java:27]: yup, just readability

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java (hunk 2 above)
Comment (jgaleotti) [reply client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java:7]: I will improve the understability of the code.

File: client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java (hunk 1 above)
Comment (jgaleotti) [reply client-java/sql/src/main/java/org/evomaster/client/java/sql/QueryResultSet.java:14]: done

**Review by arcuri82 — APPROVED** [review]

**Pull request merged**